tests/unit/test_general_functions.py

```diff
@@ -523,3 +526,35 @@ def test_all_digits_in_string_are_superscripted(self):
         actual = gf.digits_to_superscript(original)
         expected = "¹²³⁴⁵⁶⁷⁸⁹⁰"
         self.assertEqual(expected, actual)
+
+
+class TestUpdatingReferenceDensity(unittest.TestCase):
+
+    # TODO: More tests
```

Comment:
Things that stil need to be tested:
- multiple layers
- multiple elements in a layer
- both at the same time

Remember to use *Run with Coverage* button to make sure that all of the lines in `ReferenceDensity` are covered by tests.